aesd-char-driver/main.c

```diff
@@ -207,12 +211,119 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
 	return newline_pos;
 }
 
+long aesd_ioctl(struct file *filp, unsigned int cmd, unsigned long arg){
+
+	// Get device
+	struct aesd_dev *dev = filp->private_data;
+	if (dev == NULL) {
+		PDEBUG("Device cannot be accessed");
+		return -EINVAL;
+	}
+	
+	if (cmd == AESDCHAR_IOCSEEKTO){
+		
+		PDEBUG("IOCTL Detected");
+		
+		// Try to lock the device mutex
+		int res = mutex_lock_interruptible(&dev->mutex);
```

Comment:
@spinoandraptos you could minimize the locking interval here to only the portion which accessed dev